CenterForOpenScience/osf.io
# [ENG-1590] Upgrading werkzeug to 1.0.0

## Purpose

Theres an error with safari users not being able to access branded domains due to cross-domain cookies. Hopefully the new version of werkzeug will fix that.

## Changes

Changing requirements.txt for the new version. Modifying website/app.py to change the reference of ProxyFix from werkzeug.contrib (which was deprecated) to werkzeug.middleware.proxy_fix.

## QA Notes

Werkzeug is used mostly by flask, therefore we need to test all flask pages to ensure nothing broke. Below is the list of things still used by flask.

-/dashboard
-/myprojects
-All of meetings
-/registeredreports/
-/preprints/
-/registries/
-/reviews/
-Enabling and disabling addons for node and user
-Comments and Citations
-Logging in, creating account, resetting and forgetting password
-User Profile 
-/search/
-View an institutional profile
-Create new node componenet
-Create/view node forks
-Create registration from project
-View a project's registrations and draft registrations
-View a projects analytics.
-Test all file operations
-Test project contributors (adding, removing)
-Test view only links
-Create project using template
-Add/remove tags

The bulk of what needs to be tested is: 1) if you can log in and log out properly, and 2) if you can move around the site properly. Be sure to test at least a few of the pages above, and move between those pages and from those pages to ember and back.

## DevNotes

Make sure to run requirements, then recreate the web container.

## Documentation

N/A

## Side Effects

Should not be

## Ticket

https://openscience.atlassian.net/browse/ENG-1590
